crates/context/config/src/client/env/config/query/application_revision.rs

```diff
@@ -31,10 +32,34 @@ impl Method<Starknet> for ApplicationRevisionRequest {
     const METHOD: &'static str = "application_revision";
 
     fn encode(self) -> eyre::Result<Vec<u8>> {
-        todo!()
+        // Split context_id into high/low parts
+        let bytes = self.context_id.as_bytes();
+        let mid_point = bytes
+            .len()
+            .checked_div(2)
+            .ok_or_else(|| eyre::eyre!("Length should be even"))?;
+        let (high_bytes, low_bytes) = bytes.split_at(mid_point);
+
+        // Convert to Felts
+        let high_felt = Felt::from_bytes_be_slice(high_bytes);
+        let low_felt = Felt::from_bytes_be_slice(low_bytes);
+
+        // Convert both Felts to bytes and concatenate
+        let mut result = Vec::new();
+        result.extend_from_slice(&high_felt.to_bytes_be());
+        result.extend_from_slice(&low_felt.to_bytes_be());
+        Ok(result)
     }
 
-    fn decode(_response: Vec<u8>) -> eyre::Result<Self::Returns> {
-        todo!()
+    fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
+        if response.is_empty() {
+            return Err(eyre::eyre!("Empty response"));
+        }
+
+        // Response should be a single u64 in the last 8 bytes of a felt
+        let revision_bytes = &response[24..32]; // Take last 8 bytes
```

Comment:
should we check we have 32 instead

```suggestion
        if response.len() != 32 {
            return Err(eyre::eyre!("Empty response"));
        }

        // Response should be a single u64 in the last 8 bytes of a felt
        let revision_bytes = &response[24..32]; // Take last 8 bytes
```

because of the index which may panic